src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -44,7 +44,7 @@ public void delete(@PathVariable Long id) throws BookNotFoundException {
     @PutMapping("/{id}")
     public Book update(@RequestBody Book book, @PathVariable Long id)
         throws BookIdMismatchException, BookNotFoundException {
-        if (book.getId() != id)
+        if (book.getId().equals(id))
```

Comment:
If you want to ask if the IDs are different and from that throw the exception. It should go as follows:
if (! book.getId (). equals (id))